saleor/order/utils.py

```diff
@@ -819,6 +819,7 @@ def create_order_discount_for_order(
     current_total: TaxedMoney = order.undiscounted_total
     currency = order.currency
 
+    # TODO zedzior: why gross price is taken?
```

Comment:
I would say it is a bug (but you can also double check this with someone). I mean, we should use .net when channel is set to .net, and use .gross when channel is seto to .gross.